### PR TITLE
[test] [gui] Add a test of drawing lines

### PR DESCRIPTION
**Pull request by Morcki**
Related issue = https://github.com/taichi-dev/taichi/issues/5917#issuecomment-1231429518

### CODE REVIEW

File: tests/python/test_ggui.py [hunk 1]
```diff
@@ -387,3 +387,48 @@ def render():
     render()
     verify_image(window.get_image_buffer_as_numpy(), 'test_set_image')
     window.destroy()
+
+
+@pytest.mark.skipif(not _ti_core.GGUI_AVAILABLE, reason="GGUI Not Available")
+@test_utils.test(arch=supported_archs)
+def test_draw_lines():
+    ti.init(arch=ti.vulkan)
```
Comment (ailzhang) [inline tests/python/test_ggui.py:395]: this is automatically done in test_utils.test
```suggestion
```

File: tests/python/test_ggui.py [hunk 2]
```diff
@@ -387,3 +387,48 @@ def render():
...
+        scene.lines(points_pos, color=(0.28, 0.68, 0.99), width=5.0)
+        canvas.scene(scene)
+
+    for _ in range(RENDER_REPEAT):
+        render()
+        window.get_image_buffer_as_numpy()
+
+    render()
+    if (platform.system() == 'Darwin'):
+        verify_image(window.get_image_buffer_as_numpy(), 'test_draw_lines.mac')
```
Comment (ailzhang) [inline tests/python/test_ggui.py:431]: Can you add a comment explaining why the rendered result differs by platform? Thanks!

**Review by ailzhang — APPROVED** [review]
LGTM!

**Review by jim19930609 — APPROVED** [review]
Interesting, looks like the behavior is platform-dependent. Is this expected?

**Comment by Morcki** [comment]
> Interesting, looks like the behavior is platform-dependent. Is this expected?

Yes, it's a future work that we should fix it on mac.

File: tests/python/test_ggui.py (hunk 2 above)
Comment (Morcki) [reply tests/python/test_ggui.py:431]: Just because of platform difference, but I think it's a bug. I will leave a `TODO` for this.

**Pull request merged**